### PR TITLE
Allocate fake federation user IDs dynamically as fixtures instead of hardcoded constants

### PR DESCRIPTION
**Pull request by leonerd**
As per Dan's comments in https://github.com/matrix-org/sytest/pull/114#discussion_r48003848

### CODE REVIEW

File: tests/50federation/00prepare.pl [hunk 1]
```diff
@@ -133,3 +133,21 @@
          Future->done(1);
       });
    };
+
+push @EXPORT, qw( federation_user_id_fixture );
+
+my $next_user_id = 0;
+
+sub federation_user_id_fixture
```
Comment (richvdh) [inline tests/50federation/00prepare.pl]: it'd be nice to add a comment saying what this is and what it returns.

**Comment by richvdh** [comment]
lgtm except the lack of comment.

File: tests/50federation/00prepare.pl (hunk 1 above)
Comment (leonerd) [inline tests/50federation/00prepare.pl]: Doc added e10fed3

**Comment by leonerd** [comment]
Jenkins found one failing test, a seemingly-unrelated typing notification one. I can't reproduce locally. Possibly a spurious failure, so I'm going to merge anyway.

**Pull request merged**